webui/react/src/components/UserAvatar.tsx

```diff
@@ -3,19 +3,36 @@ import React from 'react';
 import Avatar, { Props as AvatarProps } from 'shared/components/Avatar';
 import useUI from 'shared/contexts/stores/UI';
 import { useUsers } from 'stores/users';
+import { DetailedUser } from 'types';
 import { Loadable } from 'utils/loadable';
 import { getDisplayName } from 'utils/user';
 
+import css from './UserAvatar.module.scss';
 export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
+  type?: 'table' | 'compactTable' | 'avatarOnly';
+  user?: DetailedUser;
   userId?: number;
 }
 
-const UserAvatar: React.FC<Props> = ({ userId, ...rest }) => {
+const UserAvatar: React.FC<Props> = ({ userId, user, type = 'avatarOnly', ...rest }) => {
   const users = Loadable.getOrElse([], useUsers()); // TODO: handle loading state
```

Comment:
This is an existing issue but we really should just take a full `DetailedUser` in as a prop instead of requiring all users to be loaded for this component to work.

Reply:
I suppose `UserAvatar` can works with any object with "username" and "displayname" attributes, right now this component is used in Table's `userRenderer` column, which is used in experiment list, model detail/registry, task list, workspace projects/list, right now APIs used to fetch table data does not return "displayname", I suppose we need to add "displayname" to these table APIs to eliminate use of "users" state in this case?

Reply:
That would be ideal but you could also just move `useUsers` up to pages and then pass the correct user in.

Reply:
I'm under the expression that we want to eliminate `useUsers` usage in this case, so that users are not required to be loaded for this components to work? Also wouldn't this create lots of duplication?

Reply:
It's not about eliminating the loading of users, it's more about making _this component_ into a pure component. It's generally better to move the data loading to the pages while having the components be pure.

That way the user avatar doesn't need a loading state and it's more portable.

Reply:
Got it